lib/core/src/Cardano/Wallet.hs

```diff
@@ -1346,28 +1346,24 @@ selectionToUnsignedTx wdrl sel s =
             fullyQualifiedWithdrawal wdrl
         }
   where
+    -- NOTE: External addresses, not known to the wallet, will be filtered out.
```

Comment:
When talking briefly with Nikola and hearing / being reminded that they will present the result of balanceTransaction in full detail to the user, this seems wrong.

We probably _should_:
1. Make the `DerivationPath` optional on `ApiCoinSelection`
2. Don't filter out pre-selected inputs/outputs
3. Ensure all other possible tx data is exposed. (We're probably missing glue code for showing datums in api here)

Reply:
(Not critical for merging this PR and I could have a look)

Reply:
Daedalus will invoke balance tx, so it will know details of external inputs already - see "inputs" in https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/balanceTransaction And in `ApiCoinSelection` we will present ADDED inps/outs/colls (always being the resource of the wallet). So the client will know everything having "inputs" and response. 
Some argument in favor of what is now is that it is in line what constructTx responds, and as both construct and balance are first step before sign+submit. It makes sense to be compatible....